circus-train-core/src/test/java/com/hotels/bdp/circustrain/core/replica/ReplicaTest.java

```diff
@@ -425,18 +440,28 @@ public void alteringExistingPartitionedReplicaTableWithPartitionsSucceeds() thro
     verify(mockMetaStoreClient).updateTableColumnStatistics(columnStatistics);
     verify(mockReplicaLocationManager).addCleanUpLocation(anyString(), any(Path.class));
     verify(mockMetaStoreClient).alter_partitions(eq(DB_NAME), eq(TABLE_NAME), alterPartitionCaptor.capture());
-    verify(mockMetaStoreClient).add_partitions(addPartitionCaptor.capture());
+    verify(mockMetaStoreClient, times(2)).add_partitions(addPartitionCaptor.capture());
 
     assertThat(alterPartitionCaptor.getValue().size(), is(1));
-    assertThat(addPartitionCaptor.getValue().size(), is(1));
+
+    // Validate that the args were expected number of batches (2), and expected batch sizes (2, then 1) since
+    // we sent a list of 3 new partitions and used a partitionBatchSize of 2
+
+    List<List<Partition>> addCaptorValues = addPartitionCaptor.getAllValues();
+    List<Partition> firstBatch = addCaptorValues.get(0);
+    List<Partition> secondBatch = addCaptorValues.get(1);
+    assertThat(addCaptorValues.size(), is(2));
+    assertThat(firstBatch.size(), is(2));
+    assertThat(secondBatch.size(), is(1));
 
     Partition altered = alterPartitionCaptor.getValue().get(0);
```

Comment:
I think to test the batching for `alter_partition` and `setPartitionColumnStatistics` usecase, it will be nice to have more than one existing partitions.

Reply:
Yeah, it feels like we are missing some tests that would have caught the issues you called out above. I'd suggest writing new tests (or modifying existing) and getting them to fail with the current code and then fixing the code and they should then pass.

Reply:
hmm, I agree with adding more tests for for `alter_partition` and `setPartitionColumnStatistics` but how would you suggest I get current(old) code to fail?  I guess I could mock a failure of the metastore client if the size of the list was longer than the `partitionBatchSize`, but that seems fairly contrived and not really testing anything except my ability to mock stuff.

Reply:
or maybe you didn't mean the issue of overwhelming the HMS with the full list, but rather the issue that I forgot to use `sublist` above?  In that case, yes, good point.

Reply:
Yes. Your last comment is what I meant. Just adding two more existing partitions in the test should give you a batch to trigger the changes you made.